src/basho_bench_worker.erl

```diff
@@ -246,30 +296,38 @@ worker_idle_loop(State) ->
 worker_next_op2(#state{api_pass_state=false}=State, OpTag) ->
     catch (State#state.driver):run(OpTag, State#state.keygen, State#state.valgen,State#state.driver_state);
 worker_next_op2(State, OpTag) ->
-   catch (State#state.driver):run(OpTag, State#state.driver_state, State).
+   OpConfig = proplists:get_value(OpTag, State#state.ops_configs),
+   KeyGen = proplists:get_value(keygen, State#state.keygen, OpConfig),
+   ValueGen = proplists:get_value(valuegen, State#state.valgen, OpConfig),
+   catch (State#state.driver):run(OpTag, KeyGen, ValueGen,
+                                  State#state.driver_state).
 
 worker_next_op(State) ->
     Next = element(random:uniform(State#state.ops_len), State#state.ops),
-    {_Label, OpTag} = Next,
+    {Label, OpTag} = Next,
```

Comment:
Skip the intermediate variable `{Label, OpTag} = element(random:uniform(State#state.ops_len), State#state.ops)`